azurelinuxagent/common/utils/processutil.py

```diff
@@ -17,11 +17,58 @@
 # Requires Python 2.6+ and Openssl 1.0+
 #
 
+from azurelinuxagent.common.exception import ExtensionError, ExtensionErrorCodes
 from azurelinuxagent.common.future import ustr
+import os
+import signal
+import subprocess
+import time
 
 TELEMETRY_MESSAGE_MAX_LEN = 3200
 
 
+def wait_for_process_completion_or_timeout(process, timeout):
+    while timeout > 0 and process.poll() is None:
+        time.sleep(1)
+        timeout -= 1
+
+    return_code = None
+
+    if timeout == 0:
+        os.killpg(os.getpgid(process.pid), signal.SIGKILL)
+    else:
+        # process completed or forked; sleep 1 sec to give the child process (if any) a chance to start
+        time.sleep(1)
+        return_code = process.wait()
+
+    return timeout == 0, return_code
+
+
+def start_subprocess_and_wait_for_completion(command, timeout, shell, cwd, env, stdout, stderr, preexec_fn, error_code):
```

Comment:
document that stdout stderr must be files (we seek on them)

Reply:
Done